hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_excl.el

```diff
@@ -38,3 +38,43 @@ ANNOTATION: "Waived the assignment as the input is wired to a parameter"
 Block 1 "3318159292" "data_o = 39'(data_i);"
 INSTANCE: tb.dut.u_tl_adapter_rom.u_tlul_data_integ_enc_instr.u_data_gen
 Block 1 "3318159292" "data_o = 39'(data_i);"
+
+INSTANCE: tb.dut.u_tl_adapter_rom
+// wr_attr_error and wr_vld_error has dependence over same type of a_opcode. Thus we can't get
+// wr_attr_error alone.
+Condition 17 "3469950311" "(wr_attr_error | wr_vld_error | rd_vld_error | instr_error | tlul_error |
+                            intg_error) 1 -1" (7 "100000")
+
+// We can't get an instr_error alone without tlul_error. Reason is same that both bits depends upon
+// instruction type being invalid.
+Condition 17 "3469950311" "(wr_attr_error | wr_vld_error | rd_vld_error | instr_error | tlul_error |
+                            intg_error) 1 -1" (4 "000100")
+
+// For rspfifo_rdata.error, it is pushed in rspfifo_wdata.error which comes from rerror_i on L529
+// in tlul_adapter_sram.sv and rerror_i is wired to 0 in rom_ctrl.sv L203
+Condition 4 "3638058042" "(rspfifo_rdata.error | reqfifo_rdata.error) 1 -1" (3 "10")
+
+// reqfifo_rvalid implies d_valid. Read from both req and rsp fifos happened on the same cycle. So
+// expecting rspfifo_rvalid without d_valid is a non-occuring case.
+Condition 21 "3623514242" "(d_valid & rspfifo_rvalid & (reqfifo_rdata.op == OpRead)) 1 -1" (1 "011")
+
+// Condition 24 till 34 are excluded for the same reason. The requirements for the conditions are:
+// 1) reqfifo_valid --> 1
+// 2) rspfifo_valid --> 1
+// 3) reqfifo_rdata.op --> opread
+// 4) refifo_rdata.error --> 1
+// For condition 24 to 26, we don't expect a rspfifo_rvalid with a reqfifo_rdata.error. Thus it
+// restrict the case 10, 1, 11 and 01 for conditions 24 till 26 as non-ocurring. For condition 34,
+// d_error needs reqfifo_rvalid and reqfifo_rdata.error to get set which automatically set the
+// d_valid. So 01 for condition 24 is non-occuring.
```

Comment:
I think this would probably be easier to follow if you wrote some text about condition 24 (following the form you've got here) and then wrote something shorter for each of 25, 26, 34 that explains why the same argument works for it too.

For condition 24, it might be something like:
```
// We cannot see vld_rd_rsp && d_error. If d_error is true, we must have rspfifo_rdata.error or
// reqfifo_rdata.error. The first cannot happen, because the items in the response fifo have an
// error signal from rerror_i[1] and rom_ctrl connects rerror_i to zero. For the second to happen,
// we must have pushed in an item with error = 1, which means that error_internal was set. But that
// will have squashed the request, so req_o was low. As a result, the request fifo item will be
// popped again before anything appears in the response fifo.
```

For condition 25, I think we're talking about the situation where `vld_rd_rsp && !reqfifo_rdata.error`? I'm slightly surprised about that! (Wouldn't we expect to see that on every read?)

For condition 26, you could refer to condition 24. Or (maybe better!) use something like what I wrote above as a description of condition 26 and then make the description of condition 24 explain why it would imply 26, which can't happen.

For condition 34, I'm a bit surprised again. Here, we're talking about `d_valid && d_error`. The argument above shows that this can only happen when we have an error in the request. But don't we see this in the TL errors sequence?

Reply:
> For condition 25, I think we're talking about the situation where vld_rd_rsp && !reqfifo_rdata.error? I'm slightly surprised about that! (Wouldn't we expect to see that on every read?)

No, the case here is to see the condition true i.e. (vld_rs_rsp && reqfifo_rdata.error) --> 1

> For condition 34, I'm a bit surprised again. Here, we're talking about d_valid && d_error. The argument above shows that this can only happen when we have an error in the request. But don't we see this in the TL errors sequence?

The case here is !d_valid && d_error.